schema/artist/highlights.js

```diff
@@ -0,0 +1,42 @@
+import { GraphQLBoolean, GraphQLObjectType, GraphQLList, GraphQLString } from "graphql"
+import { assign } from "lodash"
+import { IDFields } from "../object_identification"
+import { PartnerType } from "../partner"
+
+const PartnerArtistHighlights = new GraphQLObjectType({
```

Comment:
Interestingly enough, there's already a [PartnerArtist](https://github.com/artsy/metaphysics/blob/1a0b095c55cc2ac9b1d3d072ac52e6b9eb83b1ff/schema/partner_artist.js#L8) type.

However, I can't reuse that here (which would be ideal since this endpoint returns partner/artist pairs). The reason is b/c it introduces a 'diamond dependency'. The artist schema includes nested highlights, which itself includes a nested artist schema object.....etc.

I've run across this type of scenario often, and don't have a great solution besides this duplication. Perhaps the endpoint should just return partners? Or, if that's all we care about, we can just make this payload resolve to an array of partners.

Reply:
They did thunk about that https://github.com/graphql/graphql-js/issues/373#issuecomment-215913280. That is, don’t define the fields at file import time, but later when all files and types have loaded.

Reply:
Thunk, diamonds... so much to learn.

I'm not so clear on what requires the nesting, but I _do_ want this to be faithful to the `PartnerArtist` type returned by the API. It has important data such as the `represented_by` flag, so it wouldn't be sufficient to return just partners.

Reply:
The nesting is b/c we are including `highlights` nested within `artist` (the example in the screenshot), and also splitting out `highlights` into its own file (for ease of coding/reasoning).

Thus, we have a `highlights` file that winds up needing an `Artist` type, and an `Artist` schema that nests `highlights`. This...confuses GraphQL, to say the least.

I'll check out thunk, thanks!

Reply:
Lol ok, yea that was dumb- these were all thunks already, I must not have been using them right before 😅 

Now just re-adding the dependencies (where all fields are defined as thunks), everything is working. Could have sworn I did that initially!

Reply:
FYI https://en.wikipedia.org/wiki/Thunk